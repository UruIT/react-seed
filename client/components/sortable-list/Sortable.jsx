import React from 'react';
import SortableList from './sortable-list';
import styles from './sortable.scss';

class Sortable extends React.Component {
	constructor(props) {
		super(props);

		this.handleChange = this.handleChange.bind(this);
		this.handleChangeChecked = this.handleChangeChecked.bind(this, 'lockAxis');
		this.handleChangeNumber = this.handleChangeNumber.bind(this);

		this.state = {
			lockAxis: '',
			transitionDuration: 400,
			pressDelay: 0,
			distance: 0
		};
	}

	handleChangeChecked(param, e) {
		const value = e.target.checked ? 'y' : '';
		this.handleChange(param, value);
	}
	handleChangeNumber(e) {
		const value = Number.parseInt(e.target.value) || 0;
		this.handleChange(e.target.name, value);
	}

	handleChange(param, value) {
		this.setState({
			[param]: value
		});
	}

	render() {
		const { lockAxis, transitionDuration, pressDelay, distance } = this.state;
		return (
			<div className={styles.sortable}>
				<label>
					<span>Lock Axis</span>
					<input
						type="checkbox"
						name="lockAxis"
						checked={lockAxis}
						onChange={this.handleChangeChecked}
					/>
				</label>
				<label>
					<span>Transition duration</span>
					<input
						type="Number"
						name="transitionDuration"
						onChange={this.handleChangeNumber}
						value={transitionDuration}
					/>
				</label>
				<label>
					<span>Press Delay</span>
					<input
						type="Number"
						name="pressDelay"
						onChange={this.handleChangeNumber}
						value={pressDelay}
					/>
				</label>
				<label>
					<span>Distance</span>
					<input
						type="Number"
						name="distance"
						onChange={this.handleChangeNumber}
						value={distance}
					/>
				</label>
				<div>
					<SortableList
						lockAxis={lockAxis}
						transitionDuration={transitionDuration}
						pressDelay={pressDelay}
						distance={distance}
					/>
				</div>
			</div>
		);
	}
}

export default Sortable;
