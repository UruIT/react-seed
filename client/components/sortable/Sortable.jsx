import React from 'react';
import SortableList from './sortable-list';
import styles from './sortable.scss';

class Sortable extends React.Component {
	constructor(props) {
		super(props);

		this.handleChange = this.handleChange.bind(this);
		this.handleChangeLockAxis = this.handleChangeChecked.bind(this, 'lockAxis');
		this.handleChangeDragHandle = this.handleChangeChecked.bind(this, 'enableDragHandle');
		this.handleChangeNumber = this.handleChangeNumber.bind(this);

		this.state = {
			lockAxis: '',
			enableDragHandle: false,
			transitionDuration: 400,
			pressDelay: 0,
			distance: 0
		};
	}

	handleChangeChecked(param, e) {
		let value = e.target.checked
		if (param === 'lockAxis'){
			value = e.target.checked ? 'y' : '';
		}
		this.handleChange(param, value);
	}
	handleChangeNumber(e) {
		const value = Number.parseInt(e.target.value) || 0;
		this.handleChange(e.target.name, value);
	}

	handleChange(param, value) {
		this.setState({
			[param]: value
		});
	}

	render() {
		const { lockAxis, transitionDuration, pressDelay, distance, enableDragHandle } = this.state;
		return (
			<div className={styles.sortable}>
				<label>
					<span>Lock Axis</span>
					<input
						type="checkbox"
						name="lockAxis"
						checked={lockAxis}
						onChange={this.handleChangeLockAxis}
					/>
				</label>
				<label>
					<span>Drag Handle</span>
					<input
						type="checkbox"
						name="useDragHandle"
						checked={enableDragHandle}
						onChange={this.handleChangeDragHandle}
					/>
				</label>
				<label>
					<span>Transition duration</span>
					<input
						type="Number"
						name="transitionDuration"
						onChange={this.handleChangeNumber}
						value={transitionDuration}
					/>
				</label>
				<label>
					<span>Press Delay</span>
					<input
						type="Number"
						name="pressDelay"
						onChange={this.handleChangeNumber}
						value={pressDelay}
					/>
				</label>
				<label>
					<span>Distance</span>
					<input
						type="Number"
						name="distance"
						onChange={this.handleChangeNumber}
						value={distance}
					/>
				</label>
				<div className={styles.list}>
					<SortableList
						lockAxis={lockAxis}
						transitionDuration={transitionDuration}
						pressDelay={pressDelay}
						distance={distance}
						enableDragHandle={enableDragHandle}
					/>
				</div>
			</div>
		);
	}
}

export default Sortable;
