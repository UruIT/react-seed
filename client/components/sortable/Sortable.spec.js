import React from 'react';
import renderer from 'react-test-renderer';

import Sortable from './Sortable';

describe('<Sortable />', () => {
	it('snapshot', () => {
		const tree = renderer.create(<Sortable />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});

