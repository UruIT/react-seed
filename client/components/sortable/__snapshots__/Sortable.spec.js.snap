// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sortable /> snapshot 1`] = `
<div
  className="sortable"
>
  <label>
    <span>
      Lock Axis
    </span>
    <input
      checked=""
      name="lockAxis"
      onChange={[Function]}
      type="checkbox"
    />
  </label>
  <label>
    <span>
      Drag Handle
    </span>
    <input
      checked={false}
      name="useDragHandle"
      onChange={[Function]}
      type="checkbox"
    />
  </label>
  <label>
    <span>
      Transition duration
    </span>
    <input
      name="transitionDuration"
      onChange={[Function]}
      type="Number"
      value={400}
    />
  </label>
  <label>
    <span>
      Press Delay
    </span>
    <input
      name="pressDelay"
      onChange={[Function]}
      type="Number"
      value={0}
    />
  </label>
  <label>
    <span>
      Distance
    </span>
    <input
      name="distance"
      onChange={[Function]}
      type="Number"
      value={0}
    />
  </label>
  <div
    className="list"
  >
    <SortableList
      distance={0}
      enableDragHandle={false}
      lockAxis=""
      pressDelay={0}
      transitionDuration={400}
    />
  </div>
</div>
`;
